src/dodal/devices/i19/blueapi_device.py

```diff
@@ -0,0 +1,102 @@
+import json
+import time
+from enum import Enum
+from typing import TypeVar
+
+from aiohttp import ClientSession
+from bluesky.protocols import Movable
+from ophyd_async.core import AsyncStatus, StandardReadable
+
+from dodal.log import LOGGER
+
+OPTICS_BLUEAPI_URL = "https://i19-blueapi.diamond.ac.uk"
+HEADERS = {"Accept": "application/json", "Content-Type": "application/json"}
+
+D = TypeVar("D")
+
+
+class HutchState(str, Enum):
+    EH1 = "EH1"
+    EH2 = "EH2"
+
+
+class OpticsBlueAPIDevice(StandardReadable, Movable[D]):
+    """General device that a REST call to the blueapi instance controlling the optics \
+    hutch running on the I19 cluster, which will evaluate the current hutch in use vs \
+    the hutch sending the request and decide if the plan will be run or not.
+
+    For details see the architecture described in \
+    https://github.com/DiamondLightSource/i19-bluesky/issues/30.
+    """
+
+    def __init__(self, name: str = "") -> None:
+        self.url = OPTICS_BLUEAPI_URL
+        self.headers = HEADERS
```

Comment:
Could: Dependency injection, might make tests easier, also means the actual URL can be defined in `dodal.beamlines.i19_1/2.py`, which I think is probably better.

```suggestion
    def __init__(self, name: str = "", url: str | None = None, headers: dict[str, str] | None = None) -> None:
        self.url = url or OPTICS_BLUEAPI_URL
        self.headers = headers or HEADERS
```

Reply:
I'm not entirely sure about this tbh, it would mean adding the option for eg. the url -which is pretty much a constant and has to be the same for both hutches anyway - to every device that inherits from this one to make it work. It's okay for the shutter - but thinking of future devices, it might end up increasing the probability of making a mistake.

Reply:
I suppose if you ever need different URLs it's a refactor anyway, was just going to parity with PVs. In a dodal beamline module you define the PV prefixes, I would expect the URL prefixes for HTTP devices in the same place. Not fussed though.